examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+    if (win) win._render(ctx, bounds);
+  };
+
+  var parent = this;
+
+  this._window = window || {};
+  this._keys = {};
+
+  this._window.addEventListener('keydown', function(event) {
+    var code = event.keyCode;
+    var keys = parent._keys;
+    var win = parent._getTopWindow();
+    if (!win) return;
+
+    if (code < 37 || code > 40) return;   // ignore everything except the arrow keys
+    event.preventDefault();
+
+    if (!keys[code]) {
+      keys[code] = { timedown: event.timeStamp };
+      win._invokeButtonCallbackByCode(code, { type: 'short' });
+    }
+
+    if (event.timeStamp - keys[code].timedown > 500 && !keys[code].longPress) {
+      keys[code].longPress = true;  // Mark that we've fired the longPress
+      win._invokeButtonCallbackByCode(code, { type: 'long' });
+    }
+  });
+
+  this._window.addEventListener('keyup', function(event) {
+    var code = event.keyCode;
+    var keys = parent._keys;
+
+    if (code < 37 || code > 40) return;   // ignore everything except the arrow keys
+
+    delete keys[code];
+  });
+
+  // Everything is an Element!
+  this.Element = function(options) {
+    this.options = options || { };
+
+    this.bounds = this.options.bounds || [0, 0, 144, 168];
+    this._backgroundColor = options.backgroundColor || rocky.GColorClear;
+    this._color = options.color || rocky.GColorBlack;
+    this.z = options.z || 0;
+
+    // Intentionally empty
+    this._render = function(ctx, bounds) { };
+
+    return this;
+  };
+
+  // A Line is an Element
+  this.Line = function(options) {
+    options = options || {};
+
+    var lineElement = new parent.Element(options);
+    lineElement._p1 = [options.bounds[0], options.bounds[1]];
+    lineElement._p2 = [
+      options.bounds[0] + options.bounds[2],
+      options.bounds[1] + options.bounds[3]
+    ];
+    lineElement._strokeWidth = options.width || 2;
+
+    lineElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_stroke_color(ctx, this._color);
+      rocky.graphics_context_set_stroke_width(ctx, this._strokeWidth);
+      rocky.graphics_draw_line(ctx, this._p1, this._p2);
+    };
+
+    return lineElement;
+  };
+
+  // A Rect is an Element
+  this.Rect = function(options) {
+    options = options || {};
+
+    var rectElement = new parent.Element(options);
+
+    rectElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_fill_color(ctx, this._backgroundColor);
+      rocky.graphics_context_set_stroke_color(ctx, this._color);
+      rocky.graphics_fill_rect(ctx, this.bounds, 0, rocky.GCornersAll);
+    };
+
+    return rectElement;
+  };
+
+  // A Circle is an Element
+  this.Circle = function(options) {
+    options = options || {};
+
+    var circleElement = new parent.Element(options);
+    circleElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_fill_color(ctx, this._backgroundColor);
+      rocky.graphics_context_set_stroke_color(ctx, rocky.GColorClear);
```

Comment:
why is this always `GColorClear` whereas the other functions respect a `options.color`?